packages/calypso-products/src/main.ts

```diff
@@ -853,7 +853,7 @@ export const chooseDefaultCustomerType = ( {
 }: {
 	currentCustomerType: string;
 	selectedPlan?: string;
-	currentPlan: { productSlug: PlanSlug };
+	currentPlan: { productSlug: PlanSlug | undefined };
```

Comment:
**Note:**
The `chooseDefaultCustomerType` was always being passed `undefined` values for `currentPlan`. Further below, there's actually a check for a falsey `currentPlan` value. I'm unsure why typescript wasn't throwing an error before, but I've explicitly added the `undefined` type here.